### ISSUE TITLE
Additional tag names for unsynchronized lyrics.

### ISSUE THREAD

**Issue by RobertZenz** [owncloud/music#1109]
Currently the lyrics of a song [are fetched from two possible tags](https://github.com/owncloud/music/blob/a309ae7da41e106e205a77779a15ae524fe6e579/lib/Utility/DetailsHelper.php#L101C1-L101C98):

```php
$lyrics = ExtractorGetID3::getFirstOfTags($data, ['unsynchronised_lyric', 'unsynced lyrics']);
```

Namely `unsynchronised_lyric` and `unsynced lyrics`. In the wild this tag is not that well specified and I've seen `unsyncedlyrics` and `unsynced_lyrics` (ignoring case here) being used in large numbers (especially on songs coming from Bandcamp and similar websites).

Would it be feasible to extend this list with these two additional tags, or should there maybe be a configuration option to allow administrators to define additional tag names from which lyrics should be drawn?

**Comment by RobertZenz** [comment]
I've added the tags in #1111 at the locations you've pointed out.

However, this only leads to the first line of the lyrics being displayed. I'm not whether that is related to these changes, to the files being FLAC that I've tested with or whether it is completely unrelated.

**Comment by paulijar** [comment]
You could test the theory above by replacing the line 
https://github.com/owncloud/music/blob/a309ae7da41e106e205a77779a15ae524fe6e579/lib/Utility/DetailsHelper.php#L194-L194
with
```
$array[$key] = \implode("\n", $value); 
```

**Comment by RobertZenz** [comment]
Yes they are, though I whipped up my own hacky solultion for a test:

```php
if (is_string($value[0])) {
    $array[$key] = "";
    
    foreach ($value as &$piece) {
        $array[$key] = $array[$key] . $piece; // Missing newline here.
    }
} else {
    $array[$key] = $value[0];
}
```

With your test-code I'm getting:

 > OCA\\Music\\Utility\\DetailsHelper::encodePictureTag(): Argument #1 ($pic) must be of type array, string given, called in ./owncloud/apps/music/lib/Utility/DetailsHelper.php on line 56 in file './owncloud/apps/music/lib/Utility/DetailsHelper.php

**Comment by paulijar** [comment]
> However, empty lines are missing from the lyrics. I wonder if this is a problem in the Tag Parser library.

Well, yes. It doesn't really make any sense for the lyrics tag to get split up as multiple tags with the same name. So ultimately, the problem is either that the tags are saved incorrectly or that the parser library parses them incorrectly. Merging those items in the application code is just a workaround.

Can you view these lyrics correctly on some other application? Could you share one sample file with this problem with me? I could take a look and report the problem for the library if it seems like a library problem.

**Comment by RobertZenz** [comment]
That's a good point. I've [created a sample FLAC file showing this](https://github.com/owncloud/music/files/13641716/test.flac.zip) with the same workflow as I usually use. Dumping it shows the lyrics in multiple "instances" of the same tag:

```
~$ metaflac --list test.flac 
METADATA block #0
  type: 0 (STREAMINFO)
  is last: false
  length: 34
  minimum blocksize: 4608 samples
  maximum blocksize: 4608 samples
  minimum framesize: 14 bytes
  maximum framesize: 16 bytes
  sample_rate: 44100 Hz
  channels: 2
  bits-per-sample: 16
  total samples: 92883456
  MD5 signature: 20f3b9fbe778848f86b32d7eec9b6c1e
METADATA block #1
  type: 4 (VORBIS_COMMENT)
  is last: false
  length: 285
  vendor string: Lavf57.71.100
  comments: 8
    comment[0]: unsynced_lyrics=Multline Line1
    comment[1]: unsynced_lyrics=Line2
    comment[2]: unsynced_lyrics=Line3 Empty Line Follows
    comment[3]: unsynced_lyrics=Line4 after empty line
    comment[4]: unsynced_lyrics=Line5
    comment[5]: custom_test=Line1
Line2

Line3 after empty
linr4
    comment[6]: title=silence
    comment[7]: encoder=Lavf57.71.100
METADATA block #2
  type: 1 (PADDING)
  is last: true
  length: 7953

```

However, at least [Deadbeef](https://deadbeef.sourceforge.io/) does merge the items correctly and displays it as one. **However**, I just noticed the Deadbeef at least **does not** store the empty lines correctly in the file, at least my version does not. It displays it correctly in its cached state but as soon as the metadata is reloaded the empty lines are missing, boomer.

The tag `custom_test` was added by me with [Puddletag](https://docs.puddletag.net/), and it behaves correctly. as one would expect.

So, I'd say that the original intent of adding more tags to search through for lyrics is working as intended. That the empty lines are missing is a bug in Deadbeef on my end, that multiple lines of the same tag are stored in multiple tags is...mh...no clue to be honest. Seems like it would be ambiguous.

Thanks for your patience on this.

**Comment by RobertZenz** [comment]
Or to sum it up: Adding the additional tags works correctly, everything else seems to be a specific problem with Deadbeef (or another part of my pipeline).

**Comment by paulijar** [comment]
Okay thanks for testing and for the info. Your PR seemed fine to me, and based on the above, no additional fixes are needed. Hence, the PR was merged.

**Comment by paulijar** [comment]
This enhancement is now released as part of Music v1.10.0.
